python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -150,6 +152,26 @@ export default function AIConfigEditor({
 
   const logEventHandler = callbacks?.logEventHandler;
 
+  const downloadCallback = callbacks?.download;
+  const onDownload = useCallback(
+    async (config: AIConfig) => {
+      if (!downloadCallback) {
+        return;
+      }
+      try {
+        await downloadCallback(config);
```

Comment:
Hmm, this is kind of strange -- here we are using the client config as the source of truth and I guess sending it to the server to initiate the download? Can't we just use the server-copy similar to share?

Reply:
r.e. using server link. If we had an endpoint for the server to give us the download we would be able to just use the download property for an anchor element: https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/download

Reply:
> endpoint for the server to give us the download

You mean give us the AIConfig from server? This is similar to the `load()` endpoint which we could re-use if desired (but we'd have to refactor it to accept sessionId to get the right AIConfig instance)

Reply:
Yes. I think this is fine for now. In gradio we can just fetch from the server and download. It won't actually be 2 hops since it's just one server request (to download the config) and then the browser -> user machine is a synchronous (local) operation, right?